query/src/interpreters/interpreter_table_drop.rs

```diff
@@ -77,6 +78,21 @@ impl Interpreter for DropTableInterpreter {
         let catalog = self.ctx.get_catalog(catalog_name)?;
         catalog.drop_table(self.plan.clone().into()).await?;
 
+        if let Some(tbl) = tbl {
+            // if `plan.all`, truncate, then purge the historical data
+            if self.plan.all {
+                // errors of truncation are ignored
+                let _ = tbl
+                    .truncate(self.ctx.clone(), TruncateTablePlan {
+                        catalog: self.plan.catalog.clone(),
+                        db: self.plan.db.clone(),
+                        table: self.plan.table.clone(),
+                        purge: true,
+                    })
+                    .await;
```

Comment:
Is there a consistency issue that `drop_table` at line 79 and `truncate()` at line 86 will take place on different tables?

Reply:
yeah, there is.  seem we need something like `catalog.drop_table_by_id` here...

Reply:
https://github.com/datafuselabs/databend/issues/5626